### ISSUE TITLE
Random crash when switching between model previews in different folders

### ISSUE THREAD

**Issue by Kizari** [Kizari/Flagrum#18]
Issue as per the title. I am unable to reliably reproduce though.

However, there is enough information in the crash log to potentially resolve this issue regardless.

```
System.InvalidOperationException
Message: The current thread is not associated with the Dispatcher. Use InvokeAsync() to switch execution to the Dispatcher when triggering rendering or component state.
Stack Trace:    at Microsoft.AspNetCore.Components.Dispatcher.AssertAccess()
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.AddToRenderQueue(Int32 componentId, RenderFragment renderFragment)
   at Microsoft.AspNetCore.Components.RenderHandle.Render(RenderFragment renderFragment)
   at Microsoft.AspNetCore.Components.ComponentBase.StateHasChanged()
   at Flagrum.Web.Components.Modals.Modal.Open() in C:\Code\Flagrum\Flagrum.Web\Components\Modals\Modal.razor:line 45
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)
   at System.Threading.QueueUserWorkItemCallback.<>c.<.cctor>b__6_0(QueueUserWorkItemCallback quwi)
   at System.Threading.ExecutionContext.RunForThreadPoolUnsafe[TState](ExecutionContext executionContext, Action`1 callback, TState& state)
   at System.Threading.QueueUserWorkItemCallback.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
   at System.Threading.Thread.StartCallback()
```

Appears to be related to the opening of a modal, although I don't recall attempting to open any at the time of the crash. Perhaps a misclick?

**Comment by Rinual** [comment]
Possibly related
```
Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
   at Flagrum.Desktop.Architecture.ViewportHelper.<>c__DisplayClass3_0.<ChangeModel>b__2(Mesh mesh)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass44_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int32 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__DisplayClass44_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int32 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
   at System.Threading.Tasks.TaskReplicator.Replica.Execute()
```

**Comment by Kizari** [comment]
Haven't had the original issue again since this issue was opened. Possibly solved already or not relevant. Second issue was caused by some models not having a UV map (???) so a generic UV map with all vertices at (0, 0) has been generated at runtime for such cases so they can load into the viewer. This issue will be fixed in Flagrum 1.3.3.
